api/views/event_view.py

```diff
@@ -13,18 +13,52 @@
 )
 
 from .team_base import *
-from ..permissions import IsUserOwnerOrReadOnly
 
 class EventViewSet(TeamRelatedViewSet):
     queryset = Event.objects.all()
     serializer_class = EventSerializer
     permission_classes = [
         permissions.IsAuthenticated,
-        IsUserOwnerOrReadOnly | IsAdminMemberOrReadOnly,
+        IsAdminMemberOrReadOnly,
         IsMemberUser,
     ]
     filter_backends = [django_filters.rest_framework.DjangoFilterBackend]
     search_fields = ['^team__name', '^title', '^description', '^description', '^event_type',
                      '^user']
     ordering_fields = ['created', 'team', 'event_type', 'user']
-    
\ No newline at end of file
+
+    @action(methods=["GET"],
+            permission_classes=[permissions.IsAuthenticated, IsMemberUser]
+    )
+    def list(self, request, pk=None):
+        serializer = self.serializer_class(self.queryset, many=True)
+        event = get_object_or_404(Event, pk=pk)
+        self.check_object_permissions(request, event)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    @action(
+        detail=True,
+        methods=["GET"],
+        permission_classes=[permissions.IsAuthenticated, IsMemberUser]
+    )
+    def retrieve(self, request, *args, **kwargs):
```

Comment:
Take a closer look at TeamRelatedViewSet. That base class inherits from the RetrieveModelMixin which defines an endpoint that will return a single record at the pk.

Reply:
Are you saying that there's no need for a retrieve function in the event viewset since it's already taken care of in retrieve model mixin?

Reply:
yes